pcb-json-to-tscircuit-soup-json/index.ts

```diff
@@ -65,42 +60,80 @@ export const convertDsnJsonToTscircuitSoupJson = (
     componentSoupElements.push(soupSourceComponent, soupPcbComponent)
 
     pcbComponentImage.pins.sort(
-      (a, b) => Number(a.pin_number) - Number(b.pin_number),
+      (a, b) => Number(a.pin_number) - Number(b.pin_number)
     )
 
     const pcbComponentSmtPads = pcbComponentImage.pins.filter(({ name }) =>
-      isSmtPad(name),
+      isSmtPad(name)
     )
     for (const { pin_number, x, y, name } of pcbComponentSmtPads) {
-      const { shape, layer, width, height, radius } = parsePinName(name)
+      const { shape, layer, width, height, diameter } = parsePadName(name)
       const soupSourcePort = Soup.source_port.parse({
         type: "source_port",
         source_port_id: `source_port_${pin_number}`,
         source_component_id: soupSourceComponent.source_component_id,
         name: pin_number,
       })
 
-      const pad = {
+      const padShapeMap = {
+        Round: "circle",
+        Oval: "circle",
+        RoundRect: "rect",
+        Rect: "rect",
+        Cust: "rect",
+      }
+      const padLayerMap = {
+        A: "top",
```

Comment:
This is what it's supposed to be, there can be no `inner` layers or up to 20 depending on the number of available layers

```suggestion
        A: ["top", "inner1", "inner2", "bottom"],
```